lib/phoenix_ecto/sql/sandbox.ex

```diff
@@ -133,12 +169,39 @@ defmodule Phoenix.Ecto.SQL.Sandbox do
   end
 
   def call(conn, %{header: header, sandbox: sandbox}) do
-    _result =
-      conn
-      |> extract_metadata(header)
-      |> allow_sandbox_access(sandbox)
+    metadata = extract_metadata(conn, header)
+    _result = allow_sandbox_access(metadata, sandbox)
+    assign(conn, :phoenix_ecto_sandbox, metadata)
+  end
 
-    conn
+  if Code.ensure_loaded?(Phoenix.LiveView) do
+    @doc """
+    Allow access to the sandbox for liveview processes.
+    """
+    def allow_ecto_sandbox(%Phoenix.LiveView.Socket{} = socket, header \\ :user_agent) do
+      %{assigns: %{phoenix_ecto_sandbox: metadata}} =
+        Phoenix.LiveView.assign_new(socket, :phoenix_ecto_sandbox, fn ->
+          if Phoenix.LiveView.connected?(socket) do
+            info = Phoenix.LiveView.get_connect_info(socket)
+
+            case header do
+              :user_agent ->
+                info.user_agent
+
+              header ->
+                Enum.find_value(info.x_headers, fn
+                  {^header, val} -> val
+                  _ -> false
+                end)
+            end
+            |> decode_metadata()
+          else
+            %{}
+          end
+        end)
```

Comment:
Great PR, thank you!

However, I would prefer if did not need this code in here. Could we expose something `allow(metadata, sandbox)`? And allow will either:

1. receive a metadata map, which it allows
2. receive a binary which we decode as metadata

This way we can move the whole assign_new and friends to the LiveView itself and not have to worry about LiveView in here. WDYT? The other benefit is that people would be able to use it from channels too.

Reply:
I guess that would work as well. I'd still document an example of the above however.

Reply:
Yes, the documentation is lovely and we should keep it!